### PR TITLE
Refactor of DataLake module into dls and dla modules

### PR DESCRIPTION
**Pull request by begoldsm**
This change does the following:
* Remove datalake module
* Create top level dls (data lake store) and dla (data lake analytics) modules
* Refactor all names of all commands to fit in these two buckets
* Update parameters to have better naming and to follow the dash pattern
* Update list jobs to follow the right logic for top and filtering.
* Add support for acl management
* Add support for file expiry
* Add and update tests
* Update readme and history files.

### CODE REVIEW

**Comment by troydai** [comment]
@begoldsm please address the style and CI failure first.

File: src/command_modules/azure-cli-dla/HISTORY.rst [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+Release History
+===============
+0.0.1b3 (2017-03-22)
++++++++++++++++++++++
+
+* Rename and separate command modules for Data Lake Analytics and Data Lake Store
+* Remove datalake package
+* rename parameters for some catalog management sub groups
+* Fix support for the job list command to allow better filtering and ordering
+
```
Comment (derekbekoe) [inline src/command_modules/azure-cli-dla/HISTORY.rst]: This history would start from afresh so I'd remove the older versions.

Or you can start azure-cli-dla from 0.0.1b1. Up to you.

File: src/command_modules/azure-cli-dla/azure/cli/command_modules/dla/custom.py [hunk 2]
```diff
@@ -4,6 +4,11 @@
 # --------------------------------------------------------------------------------------------
 import time
 import uuid
+# to ensure we have consistent access to the quote() method
+try:
+    from urllib import parse as safe_url_lib
+except ImportError:
+    import urllib as safe_url_lib
```
Comment (derekbekoe) [inline src/command_modules/azure-cli-dla/azure/cli/command_modules/dla/custom.py]: The `six` compatibility library should handle this.
We use this elsewhere in the CLI.
https://pythonhosted.org/six/#module-six.moves.urllib.parse

File: src/command_modules/azure-cli-dls/azure/cli/command_modules/dls/_params.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+# ARGUMENT DEFINITIONS
+# pylint: disable=line-too-long
+datalake_store_name_type = CliArgumentType(help='Name of the Data Lake Store account.', options_list=('--account_name',), completer=get_resource_name_completion_list('Microsoft.DataLakeStore/accounts'), id_part='name')
+
+# PARAMETER REGISTRATIONS
+# global params
+register_cli_argument('dls', 'account_name', datalake_store_name_type, options_list=('--account', '-n'))
+register_cli_argument('dls', 'top', help='Maximum number of items to return.', type=int)
+register_cli_argument('dls', 'skip', help='The number of items to skip over before returning elements. Optional.', type=int)
+register_cli_argument('dls', 'count', help='The Boolean value of true or false to request a count of the matching resources included with the resources in the response, e.g. Categories?$count=true. Optional.', type=bool)
```
Comment (derekbekoe) [inline src/command_modules/azure-cli-dls/azure/cli/command_modules/dls/_params.py]: You don't need to add `Optional.` to the help strings. The CLI adds `[Required]` to arguments that are required automatically.

File: src/command_modules/azure-cli-dla/HISTORY.rst (hunk 1 above)
Comment (begoldsm) [inline src/command_modules/azure-cli-dla/HISTORY.rst]: Fair enough, I will update to go back in time to the 1b1 and remove "stale" history and the "remove datalake package" bullet.

File: src/command_modules/azure-cli-dla/azure/cli/command_modules/dla/custom.py (hunk 2 above)
Comment (begoldsm) [inline src/command_modules/azure-cli-dla/azure/cli/command_modules/dla/custom.py]: @derekbekoe interesting! How do I use it for urllib? I am confused by the documentation there what the actual module is called in six, and VS intellisense is complaining that there is no urlib.parse.quote() functionality anywhere in this module.

File: src/command_modules/azure-cli-dla/azure/cli/command_modules/dla/_params.py [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+from argcomplete.completers import FilesCompleter
+from azure.cli.core.commands import \
+    (register_cli_argument, CliArgumentType)
+from azure.cli.core.commands.parameters import \
+    (tags_type,
+     get_resource_name_completion_list,
+     resource_group_name_type,
+     enum_choice_list)
+
+from azure.cli.command_modules.dla._validators import (validate_resource_group_name,
```
Comment (troydai) [inline src/command_modules/azure-cli-dla/azure/cli/command_modules/dla/_params.py]: @derekbekoe will this introduce performance issue? The validator module in terms import msrest and other azure modules.

**Comment by troydai** [comment]
I don't have much concerns regarding the changes except the one about performance. It's better for @derekbekoe to answer.

File: src/command_modules/azure-cli-dla/azure/cli/command_modules/dla/_params.py (hunk 4 above)
Comment (begoldsm) [inline src/command_modules/azure-cli-dla/azure/cli/command_modules/dla/_params.py]: @derekbekoe and @troydai I am following a validator pattern for iso date time strings that is also in some other validators, so this should be ok (see the batch validators). Also, here are the current load times for DLA and DLS:
Loaded module 'dla' in 0.418 seconds.
Loaded module 'dls' in 0.480 seconds.

vs batch: Loaded module 'batch' in 1.096 seconds.

A lot of modules load much faster, but it looks like we are right around the middle of the pack now.

Comment (derekbekoe) [inline src/command_modules/azure-cli-dla/azure/cli/command_modules/dla/_params.py]: This LGTM.
The import of `_params.py` only happens when the command in `dla` module has been found so it wouldn't impact other modules.

**Pull request merged**